DOCS-11257-aggregator-timebase-example-kt (#1508)
mulesoft/docs-connectors: aggregators/1.0/modules/ROOT/pages/aggregators-time-example.adoc

= Time-Based Aggregator Example - Mule 4

> 

Reviewer: Edit instructions: This is pretty similar to the previous PR you review yesterday, so you can just focus on the intro paragraph and the last paragraph after the Console logger message.

> The Aggregators module *Time based aggregator* scope enables you to aggregate elements until a pre-defined time period is completed. The following example shows how to configure this scope to aggregate HTTP request calls performed within `5` seconds. Then, the application transforms the calls via DataWeave into random ID numbers. When the aggregation reaches `5` seconds, an *Aggregator listener* source gets the list of elements and the application logs all the aggregated random ID numbers.

Reviewer: ```suggestion
The Aggregators module *Time based aggregator* scope enables you to aggregate elements within a predefined time limit. The following example shows how to configure this scope to aggregate HTTP request calls within `5` seconds, followed by showing how the application transforms the calls via DataWeave into random ID numbers. When the aggregation reaches `5` seconds, an *Aggregator listener* source gets the list of elements and the application logs all the aggregated random ID numbers.
```

> .Tize-Based Aggregator Flow Example

Reviewer: ```suggestion
.Time-Based Aggregator Flow Example
```

INFO  2021-06-09 12:02:06,071 [[MuleRuntime].uber.04: [aggregator-time-demo].aggregator-time-demoFlow.CPU_INTENSIVE @59b15997] [processor: aggregator-time-demoFlow/processors/1/route/0/processors/0; event: a7cd0b40-c933-11eb-a825-f01898ad2638] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: Doing incremental step.
INFO  2021-06-09 12:02:09,206 [[MuleRuntime].uber.04: [aggregator-time-demo].aggregator-time-demoOnCompleteFlow.CPU_LITE @2ab3c271] [processor: aggregator-time-demoOnCompleteFlow/processors/0; event: 827c9591-c933-11eb-a825-f01898ad2638] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: Aggregation complete
INFO  2021-06-09 12:02:09,208 [[MuleRuntime].uber.04: [aggregator-time-demo].aggregator-time-demoOnCompleteFlow.CPU_LITE @2ab3c271] [processor: aggregator-time-demoOnCompleteFlow/processors/1; event: 827c9591-c933-11eb-a825-f01898ad2638] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: [
  {
    "Source Name": "time-based",
    "Source ID": "0.7095971146328517"
  },
  {
    "Source Name": "time-based",
    "Source ID": "0.6900116288202388"
  }
]
```

> From the previous logger message, you can identify that for each HTTP request, the *Logger* component in the *Incremental aggregator* route logs the message:

Reviewer: ```suggestion
This example logger message shows that for each HTTP request, the *Logger* component in the *Incremental aggregator* route logs the message 
```

> `Doing incremental aggregation step.` After the `timeBasedAggregator` that is referenced by the *Aggregator listener* source completes the incremental aggregations within `5` seconds, the listener is triggered with a list of all the elements. The subsequent *Logger* component logs the message `Aggregation complete` and the second *Logger* component returns the aggregated payload elements, which are requests transformed into random ID numbers:

Reviewer: ```suggestion
`Doing incremental aggregation step.` After the `timeBasedAggregator` value that is referenced by the *Aggregator listener* source completes the incremental aggregations within `5` seconds, the listener is triggered with a list of all the elements. The subsequent *Logger* component logs the message `Aggregation complete`, and the second *Logger* component returns the aggregated payload elements, which are requests transformed into random ID numbers:
```
